plugins/modules/azure_rm_sqlserver.py

```diff
@@ -277,16 +298,55 @@ def __init__(self):
         self.name = None
         self.parameters = dict()
         self.tags = None
+        # Managed Identity
+        self.identity = None
 
         self.results = dict(changed=False)
         self.state = None
         self.to_do = Actions.NoAction
         self.change_admin_password = False
+        self._managed_identity = None
 
         super(AzureRMSqlServer, self).__init__(derived_arg_spec=self.module_arg_spec,
                                                supports_check_mode=True,
                                                supports_tags=True)
 
+    @property
+    def managed_identity(self):
+        if not self._managed_identity:
+            self._managed_identity = {"identity": ResourceIdentity,
+                                      "user_assigned": UserIdentity
+                                      }
+        return self._managed_identity
+
+    def validate_identity_parameter(self):
```

Comment:
```suggestion
    def assert_identity_params(self):
```
This function fails in case that the parameters are wrong, so maybe this name is more appropriate
Just a suggestion, Up to you

Reply:
I think using assert in the name would imply that it would be optimized out with -O.  From my understanding of best security practices we should not use assert statements in general code, only in test cases.